pkg/util/net_test.go

```diff
@@ -92,3 +101,925 @@ func TestCheckCIDRSpec(t *testing.T) {
+	for _, c := range tests {
+		t.Run(c.name, func(t *testing.T) {
+			ans := GenerateMac()
+			if _, err := net.ParseMAC(ans); err != nil {
+				t.Fatalf("%v expected %v, but %v got",
+					c.name, c.want, ans)
+			}
+		})
+	}
+}
+
+func TestIp2BigInt(t *testing.T) {
+	tests := []struct {
+		expect *big.Int
+		ip     string
+		name   string
+	}{
+		{
+			name:   "correctv4",
+			ip:     "192.168.1.1",
+			expect: big.NewInt(0).SetBytes(net.ParseIP("192.168.1.1").To4()),
+		},
+		{
+			name:   "v6",
+			ip:     "1050:0:0:0:5:600:300c:326b",
+			expect: big.NewInt(0).SetBytes(net.ParseIP("1050:0:0:0:5:600:300c:326b").To16()),
+		},
+	}
+	for _, c := range tests {
+		t.Run(c.name, func(t *testing.T) {
+			if ans := Ip2BigInt(c.ip); !reflect.DeepEqual(ans, c.expect) {
+				t.Fatalf("%v expected %v, but %v got",
+					c.ip, c.expect, ans)
+			}
+		})
+	}
+}
+
+func TestBigInt2Ip(t *testing.T) {
+	tests := []struct {
+		name   string
+		ip     *big.Int
+		expect string
+	}{
+		{
+			name:   "correct",
+			ip:     big.NewInt(0).SetBytes(net.ParseIP("192.168.1.1").To4()),
+			expect: "192.168.1.1",
+		},
+		{
+			name:   "v6",
+			expect: "1050::5:600:300c:326b",
+			ip:     big.NewInt(0).SetBytes(net.ParseIP("1050:0:0:0:5:600:300c:326b").To16()),
+		},
+	}
+	for _, c := range tests {
+		t.Run(c.name, func(t *testing.T) {
+			if ans := BigInt2Ip(c.ip); !reflect.DeepEqual(ans, c.expect) {
+				t.Fatalf("%v expected %v, but %v got",
+					c.ip, c.expect, ans)
+			}
+		})
+	}
+}
+
+func TestSubnetNumber(t *testing.T) {
+	tests := []struct {
+		name   string
+		expect string
+		subnet string
+	}{
+		{
+			name:   "correct",
+			subnet: "192.168.0.1/24",
+			expect: "192.168.0.0",
+		},
+	}
+	for _, c := range tests {
+		t.Run(c.name, func(t *testing.T) {
+			if ans := SubnetNumber(c.subnet); !reflect.DeepEqual(ans, c.expect) {
+				t.Fatalf("%v expected %v, but %v got",
+					c.subnet, c.expect, ans)
+			}
+		})
+	}
+}
+
+func TestSubnetBroadcast(t *testing.T) {
+	tests := []struct {
+		name   string
+		subnet string
+		expect string
+	}{
+		{
+			name:   "correct",
+			subnet: "192.128.23.1/15",
+			expect: "192.129.255.255",
+		},
+		{
+			name:   "v6",
+			subnet: "ffff:ffff:ffff:ffff:ffff:0:ffff:ffff/96",
+			expect: "ffff:ffff:ffff:ffff:ffff:0:ffff:ffff",
+		},
+	}
+	for _, c := range tests {
+		t.Run(c.name, func(t *testing.T) {
+			if ans := SubnetBroadcast(c.subnet); !reflect.DeepEqual(ans, c.expect) {
+				t.Fatalf("%v expected %v, but %v got",
+					c.subnet, c.expect, ans)
+			}
+		})
+	}
+}
+
+func TestFirstIP(t *testing.T) {
+	tests := []struct {
+		name   string
+		subnet string
+		expect string
+		err    error
+	}{
+		{
+			name:   "correct",
+			subnet: "192.168.0.23/24",
+			expect: "192.168.0.1",
+			err:    nil,
+		},
+		{
+			name:   "controversy",
+			subnet: "192.168.0.23/32",
+			expect: "192.168.0.24",
+			err:    nil,
+		},
+		{
+			name:   "subneterr",
+			subnet: "192.168.0.0",
+			expect: "192.168.0.1",
+			err:    errors.New(""),
```

Comment:
error value should also be tested

Reply:
now all error message is included in the unitest